File: app/src/main/java/com/showdown/ds/VulkanSurfaceView.java
package com.showdown.ds;

import android.content.Context;
import android.view.SurfaceHolder;
import android.view.SurfaceView;

public final class VulkanSurfaceView extends SurfaceView implements SurfaceHolder.Callback {
    private long surfaceId;

    public VulkanSurfaceView(Context context) {
        super(context);
        getHolder().addCallback(this);
        setFocusable(true);
    }

    @Override
    public void surfaceCreated(SurfaceHolder holder) {
        surfaceId = MainActivity.nativeAttachSurface(holder.getSurface());
    }

    @Override
    public void surfaceChanged(SurfaceHolder holder, int format, int width, int height) {
    }

    @Override
    public void surfaceDestroyed(SurfaceHolder holder) {
        if (surfaceId != 0L) {
            MainActivity.nativeDetachSurface(surfaceId);
            surfaceId = 0L;
        }
    }
}

File: app/src/main/kotlin/com/showdown/ds/VulkanSurfaceView.kt
package com.showdown.ds

import android.content.Context
import android.view.SurfaceHolder
import android.view.SurfaceView

class VulkanSurfaceView(context: Context) : SurfaceView(context), SurfaceHolder.Callback {
    private var surfaceId = 0L

    init {
        holder.addCallback(this)
        isFocusable = true
    }

    override fun surfaceCreated(holder: SurfaceHolder) {
        surfaceId = MainActivity.nativeAttachSurface(holder.surface)
    }

    override fun surfaceChanged(holder: SurfaceHolder, format: Int, width: Int, height: Int) = Unit

    override fun surfaceDestroyed(holder: SurfaceHolder) {
        if (surfaceId != 0L) {
            MainActivity.nativeDetachSurface(surfaceId)
            surfaceId = 0L
        }
    }
}
